modin/core/io/text/csv_glob_dispatcher.py

```diff
@@ -239,12 +239,23 @@ def _read(cls, filepath_or_buffer, **kwargs):
             # we are not guaranteed to have the correct information regarding how many
             # columns are on each partition.
             column_widths = None
-            # Check if is list of lists
-            if isinstance(parse_dates, list) and isinstance(parse_dates[0], list):
-                for group in parse_dates:
-                    new_col_name = "_".join(group)
-                    column_names = column_names.drop(group).insert(0, new_col_name)
-            # Check if it is a dictionary
+            if isinstance(parse_dates, list):
+                for date in parse_dates:
+                    # Lists within the parse_dates list are sequences of
+                    # CSV columns that are parsed together as a single date
+                    # column. They can be a list of either string column names
+                    # or integer column indices. e.g. if parse_dates is
+                    # [[1, 2]] and columns at indices 1 and 2 are "b" and "c",
+                    # the output dataframe has the single date column "b_c". If
+                    # parse_dates is [["a", 1]] and the column at index 1 is
+                    # named "b", the output dataframe has the single date
+                    # column "a_b".
+                    if isinstance(date, list):
+                        for i, part in enumerate(date):
+                            if not isinstance(part, str):
```

Comment:
instead of doing this conversion when `part` is not a `str`, would it be safer to only do this conversion if `part` is of a certain type, such as `int`?

Reply:
Yes, done.